{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Import Libraries\n",
    "import subprocess\n",
    "import re\n",
    "import pandas as pd\n",
    "import os\n",
    "import collections\n",
    "import scipy.stats\n",
    "import numpy as np\n",
    "import scipy\n",
    "import csv\n",
    "import math\n",
    "import seaborn as sns\n",
    "import statsmodels.api as sm\n",
    "\n",
    "from functools import reduce\n",
    "from tqdm import tqdm\n",
    "from matplotlib import pyplot as plt\n",
    "%matplotlib inline\n",
    "from pandas.tools.plotting import table\n",
    "from scipy import interp\n",
    "from sklearn import svm\n",
    "from sklearn.svm import SVC\n",
    "from sklearn.datasets import make_blobs, make_classification\n",
    "\n",
    "\n",
    "from sklearn.feature_selection import SelectKBest, chi2, VarianceThreshold\n",
    "from sklearn.model_selection import RepeatedStratifiedKFold, StratifiedKFold, GridSearchCV, LeaveOneOut, train_test_split, StratifiedShuffleSplit\n",
    "from sklearn.metrics import accuracy_score, precision_score, recall_score, confusion_matrix, auc, roc_curve, precision_recall_curve, f1_score, classification_report\n",
    "from sklearn.preprocessing import StandardScaler, MinMaxScaler\n",
    "from sklearn.ensemble import RandomForestClassifier, AdaBoostClassifier, ExtraTreesClassifier\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.neighbors import KNeighborsClassifier  \n",
    "from sklearn.tree import DecisionTreeClassifier\n",
    "from sklearn.utils.multiclass import unique_labels\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.manifold import TSNE\n",
    "from mpl_toolkits.mplot3d import Axes3D"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creazione dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "string_ss=\"som_mut/ss_6_32/files/\"\n",
    "string_sl=\"som_mut/sl_32/files/\"\n",
    "string_res=\"som_mut/res/files/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "path=\"clintab_GMQL/\"+string_res\n",
    "names=[]\n",
    "barcodes_res=[]\n",
    "filelist=os.listdir(path)\n",
    "filelist.sort()\n",
    "for file in filelist:\n",
    "\n",
    "    if file.endswith(\".meta\"):\n",
    "        tmp=pd.read_csv(path+file, sep='\\t', header=None, index_col=0)\n",
    "        tmp=tmp.T\n",
    "        if tmp['biospecimen__shared__bcr_patient_barcode'].values[0] in barcodes_res:\n",
    "            print(file)\n",
    "        else:\n",
    "            names.append(file[:11])\n",
    "            barcodes_res.append(tmp['biospecimen__shared__bcr_patient_barcode'].values[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>16648965</td>\n",
       "      <td>16648965</td>\n",
       "      <td>MST1P2</td>\n",
       "      <td>11209</td>\n",
       "      <td>RNA</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>35736524</td>\n",
       "      <td>35736524</td>\n",
       "      <td>CLSPN</td>\n",
       "      <td>63967</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>88832840</td>\n",
       "      <td>88832840</td>\n",
       "      <td>PKN2</td>\n",
       "      <td>5586</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>241872095</td>\n",
       "      <td>241872095</td>\n",
       "      <td>EXO1</td>\n",
       "      <td>9156</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R_00000</td>\n",
       "      <td>2</td>\n",
       "      <td>201880094</td>\n",
       "      <td>201880094</td>\n",
       "      <td>CDK15</td>\n",
       "      <td>65061</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         0  1          2          3       4      5                  6  7\n",
       "0  R_00000  1   16648965   16648965  MST1P2  11209                RNA  C\n",
       "1  R_00000  1   35736524   35736524   CLSPN  63967  Missense_Mutation  C\n",
       "2  R_00000  1   88832840   88832840    PKN2   5586  Missense_Mutation  A\n",
       "3  R_00000  1  241872095  241872095    EXO1   9156  Missense_Mutation  A\n",
       "4  R_00000  2  201880094  201880094   CDK15  65061             Silent  C"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path=\"clintab_GMQL/\"+string_res\n",
    "\n",
    "col=['patient', 'chrom', 'start', 'stop', 'gene_symbol', 'entrez_id', 'variant_class', 'ref_allele']\n",
    "r=[]\n",
    "for i in range(len(names)):\n",
    "    tmp=pd.read_csv(path+names[i], sep='\\t', header=None)\n",
    "    tmp=tmp.values\n",
    "    p='R_00'\n",
    "    if i < 10:\n",
    "        p=p+'00'+str(i)\n",
    "    elif i< 100:\n",
    "        p=p+'0'+str(i)\n",
    "    else:\n",
    "            p=p+str(i)\n",
    "    for j in range(tmp.shape[0]):\n",
    "        chrom=tmp[j][0].replace(\"chr\", \"\")\n",
    "        r.append([p, chrom, tmp[j][1]+1, tmp[j][2], tmp[j][4], tmp[j][5], tmp[j][6], tmp[j][8]])\n",
    "\n",
    "        \n",
    "data_r=pd.DataFrame(r)\n",
    "data_r.to_csv(path_or_buf ='clintab_GMQL/som_mut/res_sm.csv', sep='\\t', header=col, index=False)\n",
    "data_r.head() #ci sono dei nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant_class</th>\n",
       "      <th>ref_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>16648965</td>\n",
       "      <td>16648965</td>\n",
       "      <td>MST1P2</td>\n",
       "      <td>11209</td>\n",
       "      <td>RNA</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>35736524</td>\n",
       "      <td>35736524</td>\n",
       "      <td>CLSPN</td>\n",
       "      <td>63967</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>88832840</td>\n",
       "      <td>88832840</td>\n",
       "      <td>PKN2</td>\n",
       "      <td>5586</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>R_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>241872095</td>\n",
       "      <td>241872095</td>\n",
       "      <td>EXO1</td>\n",
       "      <td>9156</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R_00000</td>\n",
       "      <td>2</td>\n",
       "      <td>201880094</td>\n",
       "      <td>201880094</td>\n",
       "      <td>CDK15</td>\n",
       "      <td>65061</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   patient chrom      start       stop gene_symbol  entrez_id  \\\n",
       "0  R_00000     1   16648965   16648965      MST1P2      11209   \n",
       "1  R_00000     1   35736524   35736524       CLSPN      63967   \n",
       "2  R_00000     1   88832840   88832840        PKN2       5586   \n",
       "3  R_00000     1  241872095  241872095        EXO1       9156   \n",
       "4  R_00000     2  201880094  201880094       CDK15      65061   \n",
       "\n",
       "       variant_class ref_allele  \n",
       "0                RNA          C  \n",
       "1  Missense_Mutation          C  \n",
       "2  Missense_Mutation          A  \n",
       "3  Missense_Mutation          A  \n",
       "4             Silent          C  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_r.columns=col\n",
    "data_r.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "path=\"clintab_GMQL/\"+string_sl\n",
    "names=[]\n",
    "barcodes_sl=[]\n",
    "filelist=os.listdir(path)\n",
    "filelist.sort()\n",
    "for file in filelist:\n",
    "\n",
    "    if file.endswith(\".meta\"):\n",
    "        tmp=pd.read_csv(path+file, sep='\\t', header=None, index_col=0)\n",
    "        tmp=tmp.T\n",
    "        if tmp['biospecimen__shared__bcr_patient_barcode'].values[0] in barcodes_sl:\n",
    "            print(file)\n",
    "        else:\n",
    "            names.append(file[:11])\n",
    "            barcodes_sl.append(tmp['biospecimen__shared__bcr_patient_barcode'].values[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SL_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>12860019</td>\n",
       "      <td>12860019</td>\n",
       "      <td>PRAMEF2</td>\n",
       "      <td>65122</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SL_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>13306742</td>\n",
       "      <td>13306742</td>\n",
       "      <td>RP11-219C24.6</td>\n",
       "      <td>0</td>\n",
       "      <td>RNA</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SL_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>17354542</td>\n",
       "      <td>17354542</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>23569</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SL_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>20483607</td>\n",
       "      <td>20483607</td>\n",
       "      <td>CAMK2N1</td>\n",
       "      <td>55450</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SL_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>23974977</td>\n",
       "      <td>23974977</td>\n",
       "      <td>SRSF10</td>\n",
       "      <td>10772</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          0  1         2         3              4      5                  6  7\n",
       "0  SL_00000  1  12860019  12860019        PRAMEF2  65122  Missense_Mutation  A\n",
       "1  SL_00000  1  13306742  13306742  RP11-219C24.6      0                RNA  A\n",
       "2  SL_00000  1  17354542  17354542          PADI4  23569  Missense_Mutation  T\n",
       "3  SL_00000  1  20483607  20483607        CAMK2N1  55450              3'UTR  G\n",
       "4  SL_00000  1  23974977  23974977         SRSF10  10772  Missense_Mutation  T"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path=\"clintab_GMQL/\"+string_sl\n",
    "\n",
    "col=['patient', 'chrom', 'start', 'stop', 'gene_symbol', 'entrez_id', 'variant_class', 'ref_allele']\n",
    "sl=[]\n",
    "for i in range(len(names)):\n",
    "    tmp=pd.read_csv(path+names[i], sep='\\t', header=None)\n",
    "    tmp=tmp.values\n",
    "    p='SL_00'\n",
    "    if i < 10:\n",
    "        p=p+'00'+str(i)\n",
    "    elif i< 100:\n",
    "        p=p+'0'+str(i)\n",
    "    else:\n",
    "            p=p+str(i)\n",
    "    for j in range(tmp.shape[0]):\n",
    "        chrom=tmp[j][0].replace(\"chr\", \"\")\n",
    "        sl.append([p, chrom, tmp[j][1]+1, tmp[j][2], tmp[j][4], tmp[j][5], tmp[j][6], tmp[j][8]])\n",
    "\n",
    "        \n",
    "data_sl=pd.DataFrame(sl)\n",
    "data_sl.to_csv(path_or_buf ='clintab_GMQL/som_mut/sl_sm.csv', sep='\\t', header=col, index=False)\n",
    "data_sl.head() #ci sono dei nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant_class</th>\n",
       "      <th>ref_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SL_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>12860019</td>\n",
       "      <td>12860019</td>\n",
       "      <td>PRAMEF2</td>\n",
       "      <td>65122</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SL_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>13306742</td>\n",
       "      <td>13306742</td>\n",
       "      <td>RP11-219C24.6</td>\n",
       "      <td>0</td>\n",
       "      <td>RNA</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SL_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>17354542</td>\n",
       "      <td>17354542</td>\n",
       "      <td>PADI4</td>\n",
       "      <td>23569</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SL_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>20483607</td>\n",
       "      <td>20483607</td>\n",
       "      <td>CAMK2N1</td>\n",
       "      <td>55450</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SL_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>23974977</td>\n",
       "      <td>23974977</td>\n",
       "      <td>SRSF10</td>\n",
       "      <td>10772</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    patient chrom     start      stop    gene_symbol  entrez_id  \\\n",
       "0  SL_00000     1  12860019  12860019        PRAMEF2      65122   \n",
       "1  SL_00000     1  13306742  13306742  RP11-219C24.6          0   \n",
       "2  SL_00000     1  17354542  17354542          PADI4      23569   \n",
       "3  SL_00000     1  20483607  20483607        CAMK2N1      55450   \n",
       "4  SL_00000     1  23974977  23974977         SRSF10      10772   \n",
       "\n",
       "       variant_class ref_allele  \n",
       "0  Missense_Mutation          A  \n",
       "1                RNA          A  \n",
       "2  Missense_Mutation          T  \n",
       "3              3'UTR          G  \n",
       "4  Missense_Mutation          T  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_sl.columns=col\n",
    "data_sl.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "path=\"clintab_GMQL/\"+string_ss\n",
    "names=[]\n",
    "barcodes_ss=[]\n",
    "filelist=os.listdir(path)\n",
    "filelist.sort()\n",
    "for file in filelist:\n",
    "\n",
    "    if file.endswith(\".meta\"):\n",
    "        tmp=pd.read_csv(path+file, sep='\\t', header=None, index_col=0)\n",
    "        tmp=tmp.T\n",
    "        if tmp['biospecimen__shared__bcr_patient_barcode'].values[0] in barcodes_ss:\n",
    "            print(file)\n",
    "        else:\n",
    "            names.append(file[:11])\n",
    "            barcodes_ss.append(tmp['biospecimen__shared__bcr_patient_barcode'].values[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>9601338</td>\n",
       "      <td>9601338</td>\n",
       "      <td>TMEM201</td>\n",
       "      <td>199953</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>13115682</td>\n",
       "      <td>13115682</td>\n",
       "      <td>HNRNPCL2</td>\n",
       "      <td>440563</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>21842274</td>\n",
       "      <td>21842274</td>\n",
       "      <td>HSPG2</td>\n",
       "      <td>3339</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>26029228</td>\n",
       "      <td>26029228</td>\n",
       "      <td>EXTL1</td>\n",
       "      <td>2134</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>35841739</td>\n",
       "      <td>35841739</td>\n",
       "      <td>AGO4</td>\n",
       "      <td>192670</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          0  1         2         3         4       5                  6  7\n",
       "0  SS_00000  1   9601338   9601338   TMEM201  199953             Silent  G\n",
       "1  SS_00000  1  13115682  13115682  HNRNPCL2  440563  Missense_Mutation  G\n",
       "2  SS_00000  1  21842274  21842274     HSPG2    3339  Missense_Mutation  G\n",
       "3  SS_00000  1  26029228  26029228     EXTL1    2134  Missense_Mutation  T\n",
       "4  SS_00000  1  35841739  35841739      AGO4  192670  Missense_Mutation  A"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path=\"clintab_GMQL/\"+string_ss\n",
    "\n",
    "col=['patient', 'chrom', 'start', 'stop', 'gene_symbol', 'entrez_id', 'variant_class', 'ref_allele']\n",
    "ss=[]\n",
    "for i in range(len(names)):\n",
    "    tmp=pd.read_csv(path+names[i], sep='\\t', header=None)\n",
    "    tmp=tmp.values\n",
    "    p='SS_00'\n",
    "    if i < 10:\n",
    "        p=p+'00'+str(i)\n",
    "    elif i< 100:\n",
    "        p=p+'0'+str(i)\n",
    "    else:\n",
    "            p=p+str(i)\n",
    "    for j in range(tmp.shape[0]):\n",
    "        chrom=tmp[j][0].replace(\"chr\", \"\")\n",
    "        ss.append([p, chrom, tmp[j][1]+1, tmp[j][2], tmp[j][4], tmp[j][5], tmp[j][6], tmp[j][8]])\n",
    "\n",
    "        \n",
    "data_ss=pd.DataFrame(ss)\n",
    "data_ss.to_csv(path_or_buf ='clintab_GMQL/som_mut/ss_sm.csv', sep='\\t', header=col, index=False)\n",
    "data_ss.head() #ci sono dei nan"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Selezione feature"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "dset_res=pd.read_csv(\"clintab_GMQL/som_mut/res_sm.csv\", delimiter='\\t')\n",
    "dset_senl=pd.read_csv(\"clintab_GMQL/som_mut/sl_sm.csv\", delimiter='\\t')\n",
    "dset_sens=pd.read_csv(\"clintab_GMQL/som_mut/ss_sm.csv\", delimiter='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patient</th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>entrez_id</th>\n",
       "      <th>variant_class</th>\n",
       "      <th>ref_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>9601338</td>\n",
       "      <td>9601338</td>\n",
       "      <td>TMEM201</td>\n",
       "      <td>199953</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>13115682</td>\n",
       "      <td>13115682</td>\n",
       "      <td>HNRNPCL2</td>\n",
       "      <td>440563</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>21842274</td>\n",
       "      <td>21842274</td>\n",
       "      <td>HSPG2</td>\n",
       "      <td>3339</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>26029228</td>\n",
       "      <td>26029228</td>\n",
       "      <td>EXTL1</td>\n",
       "      <td>2134</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>35841739</td>\n",
       "      <td>35841739</td>\n",
       "      <td>AGO4</td>\n",
       "      <td>192670</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>36093176</td>\n",
       "      <td>36093176</td>\n",
       "      <td>ADPRHL2</td>\n",
       "      <td>54936</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>52911823</td>\n",
       "      <td>52911823</td>\n",
       "      <td>ECHDC2</td>\n",
       "      <td>55268</td>\n",
       "      <td>Intron</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>60040032</td>\n",
       "      <td>60040032</td>\n",
       "      <td>C1orf87</td>\n",
       "      <td>127795</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>113981786</td>\n",
       "      <td>113981786</td>\n",
       "      <td>OLFML3</td>\n",
       "      <td>56944</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>150471326</td>\n",
       "      <td>150471326</td>\n",
       "      <td>RPRD2</td>\n",
       "      <td>23248</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>150966555</td>\n",
       "      <td>150966555</td>\n",
       "      <td>CERS2</td>\n",
       "      <td>29956</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>151176753</td>\n",
       "      <td>151176753</td>\n",
       "      <td>VPS72</td>\n",
       "      <td>6944</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>155204875</td>\n",
       "      <td>155204875</td>\n",
       "      <td>THBS3</td>\n",
       "      <td>7059</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>158091521</td>\n",
       "      <td>158091521</td>\n",
       "      <td>KIRREL</td>\n",
       "      <td>55243</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>159440440</td>\n",
       "      <td>159440440</td>\n",
       "      <td>OR10J1</td>\n",
       "      <td>26476</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>159887940</td>\n",
       "      <td>159887940</td>\n",
       "      <td>CFAP45</td>\n",
       "      <td>25790</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>161049083</td>\n",
       "      <td>161049083</td>\n",
       "      <td>ARHGAP30</td>\n",
       "      <td>257106</td>\n",
       "      <td>Silent</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>161052431</td>\n",
       "      <td>161052431</td>\n",
       "      <td>ARHGAP30</td>\n",
       "      <td>257106</td>\n",
       "      <td>Intron</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>181651411</td>\n",
       "      <td>181651411</td>\n",
       "      <td>CACNA1E</td>\n",
       "      <td>777</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>185221912</td>\n",
       "      <td>185221912</td>\n",
       "      <td>SWT1</td>\n",
       "      <td>54823</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>203225059</td>\n",
       "      <td>203225059</td>\n",
       "      <td>CHIT1</td>\n",
       "      <td>1118</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>223802827</td>\n",
       "      <td>223802827</td>\n",
       "      <td>TP53BP2</td>\n",
       "      <td>7159</td>\n",
       "      <td>Silent</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>1</td>\n",
       "      <td>226596892</td>\n",
       "      <td>226596892</td>\n",
       "      <td>C1orf95</td>\n",
       "      <td>375057</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>2</td>\n",
       "      <td>30966224</td>\n",
       "      <td>30966224</td>\n",
       "      <td>GALNT14</td>\n",
       "      <td>79623</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>2</td>\n",
       "      <td>38986074</td>\n",
       "      <td>38986074</td>\n",
       "      <td>SOS1</td>\n",
       "      <td>6654</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>2</td>\n",
       "      <td>42656237</td>\n",
       "      <td>42656237</td>\n",
       "      <td>MTA3</td>\n",
       "      <td>57504</td>\n",
       "      <td>Silent</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>2</td>\n",
       "      <td>85542598</td>\n",
       "      <td>85542598</td>\n",
       "      <td>MAT2A</td>\n",
       "      <td>4144</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>2</td>\n",
       "      <td>85696016</td>\n",
       "      <td>85696016</td>\n",
       "      <td>GNLY</td>\n",
       "      <td>10578</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>2</td>\n",
       "      <td>108763624</td>\n",
       "      <td>108763624</td>\n",
       "      <td>RANBP2</td>\n",
       "      <td>5903</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>SS_00000</td>\n",
       "      <td>2</td>\n",
       "      <td>147923098</td>\n",
       "      <td>147923098</td>\n",
       "      <td>ACVR2A</td>\n",
       "      <td>92</td>\n",
       "      <td>Silent</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17364</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>16</td>\n",
       "      <td>3810725</td>\n",
       "      <td>3810726</td>\n",
       "      <td>CREBBP</td>\n",
       "      <td>1387</td>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17365</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>16</td>\n",
       "      <td>20555316</td>\n",
       "      <td>20555316</td>\n",
       "      <td>ACSM2B</td>\n",
       "      <td>348158</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17366</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>16</td>\n",
       "      <td>23635090</td>\n",
       "      <td>23635090</td>\n",
       "      <td>PALB2</td>\n",
       "      <td>79728</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17367</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>16</td>\n",
       "      <td>50677472</td>\n",
       "      <td>50677472</td>\n",
       "      <td>SNX20</td>\n",
       "      <td>124460</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17368</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>16</td>\n",
       "      <td>52463953</td>\n",
       "      <td>52463953</td>\n",
       "      <td>TOX3</td>\n",
       "      <td>27324</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17369</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>16</td>\n",
       "      <td>55489772</td>\n",
       "      <td>55489772</td>\n",
       "      <td>MMP2</td>\n",
       "      <td>4313</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17370</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>16</td>\n",
       "      <td>89222117</td>\n",
       "      <td>89222117</td>\n",
       "      <td>ZNF778</td>\n",
       "      <td>197320</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17371</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>17</td>\n",
       "      <td>7674218</td>\n",
       "      <td>7674218</td>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17372</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>17</td>\n",
       "      <td>11905662</td>\n",
       "      <td>11905662</td>\n",
       "      <td>DNAH9</td>\n",
       "      <td>1770</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17373</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>17</td>\n",
       "      <td>39519980</td>\n",
       "      <td>39519980</td>\n",
       "      <td>CDK12</td>\n",
       "      <td>51755</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17374</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>17</td>\n",
       "      <td>50607900</td>\n",
       "      <td>50607901</td>\n",
       "      <td>CACNA1G</td>\n",
       "      <td>8913</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>TG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17375</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>17</td>\n",
       "      <td>58511384</td>\n",
       "      <td>58511384</td>\n",
       "      <td>MTMR4</td>\n",
       "      <td>9110</td>\n",
       "      <td>Intron</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17376</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>18</td>\n",
       "      <td>3879507</td>\n",
       "      <td>3879507</td>\n",
       "      <td>DLGAP1</td>\n",
       "      <td>9229</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17377</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>19</td>\n",
       "      <td>7911424</td>\n",
       "      <td>7911424</td>\n",
       "      <td>MAP2K7</td>\n",
       "      <td>5609</td>\n",
       "      <td>Intron</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17378</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>19</td>\n",
       "      <td>8950858</td>\n",
       "      <td>8950861</td>\n",
       "      <td>MUC16</td>\n",
       "      <td>94025</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>TCTC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17379</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>19</td>\n",
       "      <td>30443979</td>\n",
       "      <td>30443979</td>\n",
       "      <td>ZNF536</td>\n",
       "      <td>9745</td>\n",
       "      <td>Silent</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17380</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>19</td>\n",
       "      <td>37363549</td>\n",
       "      <td>37363549</td>\n",
       "      <td>HKR1</td>\n",
       "      <td>284459</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17381</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>20</td>\n",
       "      <td>45878575</td>\n",
       "      <td>45878575</td>\n",
       "      <td>ZSWIM3</td>\n",
       "      <td>140831</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17382</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>21</td>\n",
       "      <td>44627890</td>\n",
       "      <td>44627890</td>\n",
       "      <td>KRTAP10-9</td>\n",
       "      <td>386676</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17383</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>284220</td>\n",
       "      <td>284220</td>\n",
       "      <td>PLCXD1</td>\n",
       "      <td>55344</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17384</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>7334089</td>\n",
       "      <td>7334089</td>\n",
       "      <td>STS</td>\n",
       "      <td>412</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17385</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>12694438</td>\n",
       "      <td>12694438</td>\n",
       "      <td>FRMPD4</td>\n",
       "      <td>9758</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17386</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>49286222</td>\n",
       "      <td>49286222</td>\n",
       "      <td>PPP1R3F</td>\n",
       "      <td>89801</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17387</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>63724650</td>\n",
       "      <td>63724650</td>\n",
       "      <td>ARHGEF9</td>\n",
       "      <td>23229</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17388</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>65523604</td>\n",
       "      <td>65523604</td>\n",
       "      <td>LAS1L</td>\n",
       "      <td>81887</td>\n",
       "      <td>Silent</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17389</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>72644425</td>\n",
       "      <td>72644425</td>\n",
       "      <td>PHKA1</td>\n",
       "      <td>5255</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17390</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>74742559</td>\n",
       "      <td>74742559</td>\n",
       "      <td>KIAA2022</td>\n",
       "      <td>340533</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17391</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>107554470</td>\n",
       "      <td>107554470</td>\n",
       "      <td>FRMPD3</td>\n",
       "      <td>84443</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17392</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>130230619</td>\n",
       "      <td>130230619</td>\n",
       "      <td>ZNF280C</td>\n",
       "      <td>55609</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17393</th>\n",
       "      <td>SS_00104</td>\n",
       "      <td>X</td>\n",
       "      <td>141698349</td>\n",
       "      <td>141698349</td>\n",
       "      <td>SPANXD</td>\n",
       "      <td>64648</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17394 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        patient chrom      start       stop gene_symbol  entrez_id  \\\n",
       "0      SS_00000     1    9601338    9601338     TMEM201     199953   \n",
       "1      SS_00000     1   13115682   13115682    HNRNPCL2     440563   \n",
       "2      SS_00000     1   21842274   21842274       HSPG2       3339   \n",
       "3      SS_00000     1   26029228   26029228       EXTL1       2134   \n",
       "4      SS_00000     1   35841739   35841739        AGO4     192670   \n",
       "5      SS_00000     1   36093176   36093176     ADPRHL2      54936   \n",
       "6      SS_00000     1   52911823   52911823      ECHDC2      55268   \n",
       "7      SS_00000     1   60040032   60040032     C1orf87     127795   \n",
       "8      SS_00000     1  113981786  113981786      OLFML3      56944   \n",
       "9      SS_00000     1  150471326  150471326       RPRD2      23248   \n",
       "10     SS_00000     1  150966555  150966555       CERS2      29956   \n",
       "11     SS_00000     1  151176753  151176753       VPS72       6944   \n",
       "12     SS_00000     1  155204875  155204875       THBS3       7059   \n",
       "13     SS_00000     1  158091521  158091521      KIRREL      55243   \n",
       "14     SS_00000     1  159440440  159440440      OR10J1      26476   \n",
       "15     SS_00000     1  159887940  159887940      CFAP45      25790   \n",
       "16     SS_00000     1  161049083  161049083    ARHGAP30     257106   \n",
       "17     SS_00000     1  161052431  161052431    ARHGAP30     257106   \n",
       "18     SS_00000     1  181651411  181651411     CACNA1E        777   \n",
       "19     SS_00000     1  185221912  185221912        SWT1      54823   \n",
       "20     SS_00000     1  203225059  203225059       CHIT1       1118   \n",
       "21     SS_00000     1  223802827  223802827     TP53BP2       7159   \n",
       "22     SS_00000     1  226596892  226596892     C1orf95     375057   \n",
       "23     SS_00000     2   30966224   30966224     GALNT14      79623   \n",
       "24     SS_00000     2   38986074   38986074        SOS1       6654   \n",
       "25     SS_00000     2   42656237   42656237        MTA3      57504   \n",
       "26     SS_00000     2   85542598   85542598       MAT2A       4144   \n",
       "27     SS_00000     2   85696016   85696016        GNLY      10578   \n",
       "28     SS_00000     2  108763624  108763624      RANBP2       5903   \n",
       "29     SS_00000     2  147923098  147923098      ACVR2A         92   \n",
       "...         ...   ...        ...        ...         ...        ...   \n",
       "17364  SS_00104    16    3810725    3810726      CREBBP       1387   \n",
       "17365  SS_00104    16   20555316   20555316      ACSM2B     348158   \n",
       "17366  SS_00104    16   23635090   23635090       PALB2      79728   \n",
       "17367  SS_00104    16   50677472   50677472       SNX20     124460   \n",
       "17368  SS_00104    16   52463953   52463953        TOX3      27324   \n",
       "17369  SS_00104    16   55489772   55489772        MMP2       4313   \n",
       "17370  SS_00104    16   89222117   89222117      ZNF778     197320   \n",
       "17371  SS_00104    17    7674218    7674218        TP53       7157   \n",
       "17372  SS_00104    17   11905662   11905662       DNAH9       1770   \n",
       "17373  SS_00104    17   39519980   39519980       CDK12      51755   \n",
       "17374  SS_00104    17   50607900   50607901     CACNA1G       8913   \n",
       "17375  SS_00104    17   58511384   58511384       MTMR4       9110   \n",
       "17376  SS_00104    18    3879507    3879507      DLGAP1       9229   \n",
       "17377  SS_00104    19    7911424    7911424      MAP2K7       5609   \n",
       "17378  SS_00104    19    8950858    8950861       MUC16      94025   \n",
       "17379  SS_00104    19   30443979   30443979      ZNF536       9745   \n",
       "17380  SS_00104    19   37363549   37363549        HKR1     284459   \n",
       "17381  SS_00104    20   45878575   45878575      ZSWIM3     140831   \n",
       "17382  SS_00104    21   44627890   44627890   KRTAP10-9     386676   \n",
       "17383  SS_00104     X     284220     284220      PLCXD1      55344   \n",
       "17384  SS_00104     X    7334089    7334089         STS        412   \n",
       "17385  SS_00104     X   12694438   12694438      FRMPD4       9758   \n",
       "17386  SS_00104     X   49286222   49286222     PPP1R3F      89801   \n",
       "17387  SS_00104     X   63724650   63724650     ARHGEF9      23229   \n",
       "17388  SS_00104     X   65523604   65523604       LAS1L      81887   \n",
       "17389  SS_00104     X   72644425   72644425       PHKA1       5255   \n",
       "17390  SS_00104     X   74742559   74742559    KIAA2022     340533   \n",
       "17391  SS_00104     X  107554470  107554470      FRMPD3      84443   \n",
       "17392  SS_00104     X  130230619  130230619     ZNF280C      55609   \n",
       "17393  SS_00104     X  141698349  141698349      SPANXD      64648   \n",
       "\n",
       "           variant_class ref_allele  \n",
       "0                 Silent          G  \n",
       "1      Missense_Mutation          G  \n",
       "2      Missense_Mutation          G  \n",
       "3      Missense_Mutation          T  \n",
       "4      Missense_Mutation          A  \n",
       "5      Missense_Mutation          C  \n",
       "6                 Intron          G  \n",
       "7      Missense_Mutation          C  \n",
       "8                  3'UTR          T  \n",
       "9      Missense_Mutation          G  \n",
       "10     Missense_Mutation          A  \n",
       "11     Missense_Mutation          G  \n",
       "12     Missense_Mutation          T  \n",
       "13     Missense_Mutation          G  \n",
       "14     Missense_Mutation          T  \n",
       "15     Missense_Mutation          A  \n",
       "16                Silent          T  \n",
       "17                Intron          C  \n",
       "18     Missense_Mutation          T  \n",
       "19     Missense_Mutation          T  \n",
       "20                Silent          G  \n",
       "21                Silent          A  \n",
       "22     Missense_Mutation          C  \n",
       "23                Silent          C  \n",
       "24     Missense_Mutation          T  \n",
       "25                Silent          T  \n",
       "26     Missense_Mutation          G  \n",
       "27     Missense_Mutation          T  \n",
       "28     Missense_Mutation          A  \n",
       "29                Silent          T  \n",
       "...                  ...        ...  \n",
       "17364    Frame_Shift_Ins          -  \n",
       "17365             Silent          C  \n",
       "17366  Missense_Mutation          T  \n",
       "17367  Missense_Mutation          A  \n",
       "17368  Missense_Mutation          T  \n",
       "17369             Silent          G  \n",
       "17370             Silent          C  \n",
       "17371  Missense_Mutation          T  \n",
       "17372  Missense_Mutation          G  \n",
       "17373  Missense_Mutation          A  \n",
       "17374    Frame_Shift_Del         TG  \n",
       "17375             Intron          C  \n",
       "17376  Missense_Mutation          G  \n",
       "17377             Intron          T  \n",
       "17378    Frame_Shift_Del       TCTC  \n",
       "17379             Silent          C  \n",
       "17380  Missense_Mutation          G  \n",
       "17381  Missense_Mutation          C  \n",
       "17382  Missense_Mutation          G  \n",
       "17383  Missense_Mutation          C  \n",
       "17384  Missense_Mutation          C  \n",
       "17385  Missense_Mutation          A  \n",
       "17386  Missense_Mutation          G  \n",
       "17387  Missense_Mutation          T  \n",
       "17388             Silent          G  \n",
       "17389  Missense_Mutation          C  \n",
       "17390  Missense_Mutation          T  \n",
       "17391  Missense_Mutation          G  \n",
       "17392  Missense_Mutation          C  \n",
       "17393  Missense_Mutation          G  \n",
       "\n",
       "[17394 rows x 8 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dset_sens"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#modifica!\n",
    "sl_RNA=dset_senl[dset_senl['variant_class']=='RNA']\n",
    "sl_silent=dset_senl[dset_senl['variant_class']=='Silent']\n",
    "sl_MM=dset_senl[dset_senl['variant_class']=='Missense_Mutation']\n",
    "sl_intron=dset_senl[dset_senl['variant_class']=='Intron']\n",
    "sl_FSD=dset_senl[dset_senl['variant_class']=='Frame_Shift_Del']\n",
    "sl_U=dset_senl[dset_senl['variant_class']=='3\\'UTR']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "ss_RNA=dset_sens[dset_sens['variant_class']=='RNA']\n",
    "ss_silent=dset_sens[dset_sens['variant_class']=='Silent']\n",
    "ss_MM=dset_sens[dset_sens['variant_class']=='Missense_Mutation']\n",
    "ss_intron=dset_sens[dset_sens['variant_class']=='Intron']\n",
    "ss_FSD=dset_sens[dset_sens['variant_class']=='Frame_Shift_Del']\n",
    "ss_FSI=dset_sens[dset_sens['variant_class']=='Frame_Shift_Ins']\n",
    "ss_U=dset_sens[dset_sens['variant_class']=='3\\'UTR']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "r_RNA=dset_res[dset_res['variant_class']=='RNA']\n",
    "r_silent=dset_res[dset_res['variant_class']=='Silent']\n",
    "r_MM=dset_res[dset_res['variant_class']=='Missense_Mutation']\n",
    "r_intron=dset_res[dset_res['variant_class']=='Intron']\n",
    "r_FSD=dset_res[dset_res['variant_class']=='Frame_Shift_Del']\n",
    "r_FSI=dset_res[dset_res['variant_class']=='Frame_Shift_Ins']\n",
    "r_U=dset_res[dset_res['variant_class']=='3\\'UTR']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(295, 8)"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sl_U.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Intersezione "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "ge1=pd.read_csv(\"new_data/gene_exprs/r_sl_bonf_ge.csv\", delimiter=',')\n",
    "ge2=pd.read_csv(\"new_data/gene_exprs/r_ss_bonf_ge.csv\", delimiter=',')\n",
    "ge3=pd.read_csv(\"new_data/gene_exprs/sl_ss_bonf_ge.csv\", delimiter=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lalli/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py:6201: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=True'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass sort=False\n",
      "\n",
      "  sort=sort)\n"
     ]
    }
   ],
   "source": [
    "r_ge=ge1.append(ge2)\n",
    "sl_ge=ge1.append(ge3)\n",
    "ss_ge=ge2.append(ge3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "lista1=set(dset_res['gene_symbol'])\n",
    "lista2=set(r_ge['gene_symbol'])\n",
    "lista3=set(dset_sens['gene_symbol'])\n",
    "lista4=set(ss_ge['gene_symbol'])\n",
    "lista5=set(dset_senl['gene_symbol'])\n",
    "lista6=set(sl_ge['gene_symbol'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "x=lista1.intersection(lista2)\n",
    "y=lista3.intersection(lista4)\n",
    "z=lista5.intersection(lista6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'SPTA1', 'SYNM', 'AARS', 'DDR1', 'HYOU1', 'MYL4', 'HES1', 'MAGI3', 'RPS24', 'HIRA', 'NUGGC', 'ZNF366', 'HSPA5', 'PDE3A', 'APLP2', 'XXbac-BPG246D15.9', 'RPL32', 'TPMT', 'TNFAIP8L3', 'KDR', 'KCTD1', 'TMEM164', 'RPL11', 'EPAS1', 'VCP', 'ADCK2'} {'SPTA1', 'SYNM', 'AARS', 'PDIA4', 'HYOU1', 'DDR1', 'SYT12', 'CBX7', 'BCL2L13', 'MYL4', 'RNF121', 'MRGPRX2', 'HIRA', 'NUGGC', 'ZNF366', 'HSPA5', 'APLP2', 'HS3ST5', 'RPL32', 'FGFBP1', 'TNFAIP8L3', 'OR5T1', 'KDR', 'C4A', 'RGP1', 'TIRAP', 'RPL11', 'LACRT', 'EPAS1', 'EXOSC1', 'ADCK2', 'UHRF1BP1', 'FZD3'} {'FZD5', 'PDE3A', 'HES1', 'PDIA4', 'OR5T1', 'MECOM', 'C4A', 'KCTD11'}\n"
     ]
    }
   ],
   "source": [
    "print(x, y, z)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "mi1=pd.read_csv(\"new_data/miRNA/r_sl_bonf_miRNA.csv\", delimiter=',')\n",
    "mi2=pd.read_csv(\"new_data/miRNA/sl_ss_bonf_miRNA.csv\", delimiter=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lalli/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py:6201: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=True'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass sort=False\n",
      "\n",
      "  sort=sort)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensemble_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>gene_symbol</th>\n",
       "      <th>mannwhiteney_pvalue</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>resistant_median</th>\n",
       "      <th>sensitive_long_median</th>\n",
       "      <th>sensitive_short_median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsa</td>\n",
       "      <td>hsa-mir-1301</td>\n",
       "      <td>mir-1301</td>\n",
       "      <td>0.000022</td>\n",
       "      <td>0.002046</td>\n",
       "      <td>67.507454</td>\n",
       "      <td>126.137998</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsa</td>\n",
       "      <td>hsa-mir-4761</td>\n",
       "      <td>mir-4761</td>\n",
       "      <td>0.000293</td>\n",
       "      <td>0.041252</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsa</td>\n",
       "      <td>hsa-mir-1301</td>\n",
       "      <td>mir-1301</td>\n",
       "      <td>0.000303</td>\n",
       "      <td>0.042682</td>\n",
       "      <td>NaN</td>\n",
       "      <td>126.137998</td>\n",
       "      <td>76.205448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsa</td>\n",
       "      <td>hsa-mir-1180</td>\n",
       "      <td>mir-1180</td>\n",
       "      <td>0.000343</td>\n",
       "      <td>0.048305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>225.496917</td>\n",
       "      <td>153.256471</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ensemble_id          gene gene_symbol  mannwhiteney_pvalue  p_value_corr  \\\n",
       "0         hsa  hsa-mir-1301    mir-1301             0.000022      0.002046   \n",
       "0         hsa  hsa-mir-4761    mir-4761             0.000293      0.041252   \n",
       "1         hsa  hsa-mir-1301    mir-1301             0.000303      0.042682   \n",
       "2         hsa  hsa-mir-1180    mir-1180             0.000343      0.048305   \n",
       "\n",
       "   resistant_median  sensitive_long_median  sensitive_short_median  \n",
       "0         67.507454             126.137998                     NaN  \n",
       "0               NaN               0.000000                0.000000  \n",
       "1               NaN             126.137998               76.205448  \n",
       "2               NaN             225.496917              153.256471  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sl_mi=mi1.append(mi2)\n",
    "sl_mi.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "lista7=set(dset_senl['gene_symbol'])\n",
    "lista8=set(sl_mi['gene'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "w=lista7.intersection(lista8)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "met1=pd.read_csv(\"new_data/methylation/r_sl_bonf_meth.csv\", delimiter=',')\n",
    "met2=pd.read_csv(\"new_data/methylation/r_ss_bonf_meth.csv\", delimiter=',')\n",
    "met3=pd.read_csv(\"new_data/methylation/sl_ss_bonf_meth.csv\", delimiter=',')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lalli/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py:6201: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=True'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass sort=False\n",
      "\n",
      "  sort=sort)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>mannwhiteney_pvalue</th>\n",
       "      <th>p_value_corr</th>\n",
       "      <th>resistant_median</th>\n",
       "      <th>sensitive_long_median</th>\n",
       "      <th>sensitive_short_median</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RP11-886P16.10</td>\n",
       "      <td>0.000221</td>\n",
       "      <td>0.036857</td>\n",
       "      <td>0.032653</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.025334</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RILP</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.001576</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.335616</td>\n",
       "      <td>0.289588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SKIV2L</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.003747</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.931839</td>\n",
       "      <td>0.894482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PLEKHB1</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>0.009131</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.568565</td>\n",
       "      <td>0.274356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>VPS37B</td>\n",
       "      <td>0.000062</td>\n",
       "      <td>0.017429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.429182</td>\n",
       "      <td>0.219542</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             gene  mannwhiteney_pvalue  p_value_corr  resistant_median  \\\n",
       "0  RP11-886P16.10             0.000221      0.036857          0.032653   \n",
       "0            RILP             0.000006      0.001576               NaN   \n",
       "1          SKIV2L             0.000009      0.003747               NaN   \n",
       "2         PLEKHB1             0.000033      0.009131               NaN   \n",
       "3          VPS37B             0.000062      0.017429               NaN   \n",
       "\n",
       "   sensitive_long_median  sensitive_short_median  \n",
       "0                    NaN                0.025334  \n",
       "0               0.335616                0.289588  \n",
       "1               0.931839                0.894482  \n",
       "2               0.568565                0.274356  \n",
       "3               0.429182                0.219542  "
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r_met=met1.append(met2)\n",
    "sl_met=met1.append(met3)\n",
    "ss_met=met2.append(met3)\n",
    "\n",
    "ss_met.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "lista_=set(r_met['gene'])\n",
    "lista_1=set(ss_met['gene'])\n",
    "lista_2=set(sl_met['gene'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "c=lista1.intersection(lista_)\n",
    "d=lista3.intersection(lista_1)\n",
    "e=lista5.intersection(lista_2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'SKIV2L', 'AGO1'} {'HSPB2', 'SKIV2L', 'SEMA3G', 'ZNF560'} {'HSPB2', 'PLCE1', 'ALPK3', 'SEMA3G', 'TGM1'}\n"
     ]
    }
   ],
   "source": [
    "print(c, d, e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "res=c.intersection(x)\n",
    "ss=d.intersection(y)\n",
    "sl=e.intersection(z & w)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "set() set() set()\n"
     ]
    }
   ],
   "source": [
    "print(res, ss, sl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
